gobblin-api/src/main/java/gobblin/converter/Converter.java

```diff
@@ -72,7 +72,8 @@ public abstract SO convertSchema(SI inputSchema, WorkUnitState workUnit)
    * @param workUnit a {@link WorkUnitState} object carrying configuration properties
    * @return converted data record
    * @throws DataConversionException if it fails to convert the input data record
+   * @throws IOException 
    */
   public abstract Iterable<DO> convertRecord(SO outputSchema, DI inputRecord, WorkUnitState workUnit)
-      throws DataConversionException;
+      throws DataConversionException, IOException;
```

Comment:
Can you catch your IOException and throw the DataConversionException only? Changing the Converter interface may not be an good idea. DataConversionException is designed for anything wrong with data conversion.